ui/src/components/CopyToClipboardTableCell/index.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from 'react';
+import { node, string } from 'prop-types';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import TableCell from '@material-ui/core/TableCell';
+import ContentCopyIcon from 'mdi-react/ContentCopyIcon';
+import CheckIcon from 'mdi-react/CheckIcon';
+import TableCellItem from '../TableCellItem';
+
+function CopyToClipboardTableCell(props) {
+  const [isCopy, setCopy] = useState(false);
+  const { tooltipTitle, textToCopy, text } = props;
+
+  function handleCopyClick() {
+    setCopy(true);
+  }
+
+  useEffect(() => {
+    if (isCopy) {
+      setTimeout(() => {
+        setCopy(false);
+      }, 3000);
+    }
+  }, [isCopy]);
+
+  const IconSize = 16;
+
+  return (
+    <CopyToClipboard
+      onCopy={handleCopyClick}
+      title={`${tooltipTitle} (${isCopy ? 'Copied!' : 'Copy'})`}
+      text={textToCopy}>
+      <TableCell>
+        <TableCellItem button>
+          {text}
+          {isCopy ? <CheckIcon /> : <ContentCopyIcon size={IconSize} />}
+        </TableCellItem>
+      </TableCell>
+    </CopyToClipboard>
+  );
+}
+
+CopyToClipboardTableCell.propTypes = {
+  // A tooltip to show when the user hovers on the list item
+  tooltipTitle: string.isRequired,
+  // Text to copy when a user clicks on the list item
+  textToCopy: string.isRequired,
+  // The TableCellText secondary prop
+  text: node,
+};
+
+CopyToClipboardTableCell.defaultProps = {
+  text: null,
+};
+
+export default CopyToClipboardTableCell;
```

Comment:
`CopyToClipboardTableCell` and `CopyToClipboardListItem` share a lot of the same code. We should try to avoid rewriting duplicate code. One way to do this is to write a react hook. For example, we can create a file `src/utils/useCopyToClipboard.js` with the following content:

```jsx
import { useEffect, useState } from 'react';

function useCopyToClipboard() {
  const [isCopy, setCopy] = useState(false);

  function handleCopyClick() {
    setCopy(true);
  }

  useEffect(() => {
    if (isCopy) {
      setTimeout(() => {
        setCopy(false);
      }, 3000);
    }
  }, [isCopy]);

  return {
    onCopyClick: handleCopyClick,
    isCopy,
  };
}

export default useCopyToClipboard;
```

We would then need to make use of this file in both files. For example, in `CopyToClipboardTableCell`, we would do:

```diff
diff --git a/ui/src/components/CopyToClipboardTableCell/index.jsx b/ui/src/components/CopyToClipboardTableCell/index.jsx
index 239b53dbf..96dd7b8f9 100644
--- a/ui/src/components/CopyToClipboardTableCell/index.jsx
+++ b/ui/src/components/CopyToClipboardTableCell/index.jsx
@@ -1,38 +1,30 @@
-import React, { useState, useEffect } from 'react';
+import React from 'react';
 import { node, string } from 'prop-types';
 import { CopyToClipboard } from 'react-copy-to-clipboard';
 import TableCell from '@material-ui/core/TableCell';
 import ContentCopyIcon from 'mdi-react/ContentCopyIcon';
 import CheckIcon from 'mdi-react/CheckIcon';
 import TableCellItem from '../TableCellItem';
+import useCopyToClipboard from '../../utils/useCopyToClipboard';
 
 function CopyToClipboardTableCell(props) {
-  const [isCopy, setCopy] = useState(false);
+  const { isCopy, onCopyClick } = useCopyToClipboard();
   const { tooltipTitle, textToCopy, text } = props;
-
-  function handleCopyClick() {
-    setCopy(true);
-  }
-
-  useEffect(() => {
-    if (isCopy) {
-      setTimeout(() => {
-        setCopy(false);
-      }, 3000);
-    }
-  }, [isCopy]);
-
-  const IconSize = 16;
+  const iconSize = 16;
 
   return (
     <CopyToClipboard
-      onCopy={handleCopyClick}
+      onCopy={onCopyClick}
       title={`${tooltipTitle} (${isCopy ? 'Copied!' : 'Copy'})`}
       text={textToCopy}>
       <TableCell>
         <TableCellItem button>
           {text}
-          {isCopy ? <CheckIcon /> : <ContentCopyIcon size={IconSize} />}
+          {isCopy ? (
+            <CheckIcon size={iconSize} />
+          ) : (
+            <ContentCopyIcon size={iconSize} />
+          )}
         </TableCellItem>
       </TableCell>
     </CopyToClipboard>
```

Reply:
> We would then need to make use of this file in both files. For example, in `CopyToClipboardTableCell`, we would do:

Should I change in the `CopyToClipboardListItem` component also

Reply:
Yep.